subworkflows/nf-core/fasta_build_add_kraken2_bracken/main.nf

```diff
@@ -0,0 +1,50 @@
+include { KRAKEN2_ADD   } from '../../../modules/nf-core/kraken2/add/main'
+include { KRAKEN2_BUILD } from '../../../modules/nf-core/kraken2/build/main'
+include { BRACKEN_BUILD } from '../../../modules/nf-core/bracken/build/main'
+
+workflow FASTA_BUILD_ADD_KRAKEN2_BRACKEN {
+
+    take:
+    ch_fasta              // channel: [ val(meta), fasta ]
+    ch_taxonomy_names     // channel: [ names.dmp ]
+    ch_taxonomy_nodes     // channel: [ nodes.dmp ]
+    ch_accession2taxid    // channel: [ acc2taxidfile ]
+    val_cleanintermediates   // value:   [ true | false ]
+    val_runbrackenbuild   // value:   [ true | false ]
+
+    main:
+
+    if ( val_cleanintermediates && val_runbrackenbuild ) { error("Cannot perform Kraken2 cleanup and build Bracken database. Bracken requires intermediate files") }
+    val_cleanup = [ val_cleanintermediates && !val_runbrackenbuild ].any() ? true : false
+
+    ch_versions = Channel.empty()
+
+    ch_fastas_for_kraken2add = ch_fasta
+                                .map {
+                                    meta, fasta ->
+
+                                    [[id: 'db'], fasta]
```

Comment:
To be fair I don't like the hardcoding of meta, but I don't think there is a better option here indeed (apart from adding an extra input value, but that's just as much effort as changing it after running the subworkflow)

Reply:
I think it would be a bit safer to do it earlier rather later, so I tihnk it's a good idea.

I might think about it though because it wouldn't work if someone was running the same subworkflow in parallel...

Reply:
Now I think about it, we can just push that grouping to pipeline level? So input is `[meta, [fasta1, fasta2, fasta3]]`, what do you think @alxndrdiaz ?

This would work much nicer, particularly if someone wanted to build multiple databases in a single pipeline run

Reply:
I like this solution! Much cleaner and more flexible :partying_face:

Reply:
> Now I think about it, we can just push that grouping to pipeline level? So input is `[meta, [fasta1, fasta2, fasta3]]`, what do you think @alxndrdiaz ?
> 
> This would work much nicer, particularly if someone wanted to build multiple databases in a single pipeline run

It's OK as long as each database has a different identifier use to name its directory.

Reply:
In this case it should be yes because it would come via the meta of the already grouped FASTAs :)

Reply:
I'll update this for both subworkflows now :)